sympy/printing/preview.py

```diff
@@ -29,15 +29,100 @@ def _check_output_no_window(*args, **kwargs):
     return check_output(*args, creationflags=creation_flag, **kwargs)
 
 
-def _run_pyglet(fname, fmt):
-    from pyglet import window, image, gl
-    from pyglet.window import key
+def _start_file(fname):
+    """
+    Like :func:`os.startfile`, but does not return until the program exits.
+
+    This is necessary because we delete the temp dir as soon as this function
+    returns
+
+    Equivalent code if we depended on ``pywin32``::
+
+        import win32con
+        import win32api
+        import win32event
+        from win32com.shell import shellcon
+        from win32com.shell.shell import ShellExecuteEx
+        rc = ShellExecuteEx(
+            fMask=shellcon.SEE_MASK_NOCLOSEPROCESS,
+            nShow=win32con.SW_SHOW,
+            lpFile=fname)
+        hproc = rc['hProcess']
+        win32event.WaitForSingleObject(hproc, win32event.INFINITE)
+        win32api.CloseHandle(hproc)
+    """
+    import ctypes
+    from ctypes.wintypes import ULONG, DWORD, HANDLE, HKEY, HINSTANCE, HWND, LPCWSTR
+    from _winapi import CloseHandle, WaitForSingleObject
+
+    class SHELLEXECUTEINFOW(ctypes.Structure):
+        _fields_ = [
+            ("cbSize", DWORD),
+            ("fMask", ULONG),
+            ("hwnd", HWND),
+            ("lpVerb", LPCWSTR),
+            ("lpFile", LPCWSTR),
+            ("lpParameters", LPCWSTR),
+            ("lpDirectory", LPCWSTR),
+            ("nShow", ctypes.c_int),
+            ("hInstApp", HINSTANCE),
+            ("lpIDList", ctypes.c_void_p),
+            ("lpClass", LPCWSTR),
+            ("hkeyClass", HKEY),
+            ("dwHotKey", DWORD),
+            ("DUMMYUNIONNAME", ctypes.c_void_p),
+            ("hProcess", HANDLE)
+        ]
+
+    shell_execute_ex = ctypes.windll.shell32.ShellExecuteExW
+    shell_execute_ex.argtypes = [ctypes.POINTER(SHELLEXECUTEINFOW)]
+    shell_execute_ex.res_type = ctypes.c_bool
+
+    # https://stackoverflow.com/a/17638969/102441
+    arg = SHELLEXECUTEINFOW()
+    arg.cbSize = ctypes.sizeof(arg)
+    arg.fMask = 0x00000040  # SEE_MASK_NOCLOSEPROCESS
+    arg.hwnd = None
+    arg.lpVerb = None
+    arg.lpFile = fname
+    arg.lpParameters = ""
+    arg.lpDirectory = None
+    arg.nShow = 10  # SW_SHOWDEFAULT
+    arg.hInstApp = None
+    ok = shell_execute_ex(arg)
+    if not ok:
+        raise ctypes.WinError()
+    proc = int(arg.hProcess)
+    try:
+        WaitForSingleObject(proc, -1)
+    finally:
+        CloseHandle(proc)
+
+
+def system_default_viewer(fname, fmt):
+    import platform
+    if platform.system() == 'Darwin':
+        import subprocess
+        subprocess.call(('open', fname))
+    elif platform.system() == 'Windows':
+        _start_file(fname)
+    else:
+        import subprocess
+        subprocess.call(('xdg-open', fname))
```

Comment:
On some Linux systems this would be `gio open fname`

Reply:
Which systems? Presumably `xdg-open` does not exist in those systems

Reply:
Looks like maybe I got confused by the same warning discussed here:
https://github.com/github/hub/issues/1473